File: backend/GameApp.Domain/ValueObjects/Scenes/ChangeBiomeScene.cs
using GameApp.Domain.Enumerates;

namespace GameApp.Domain.ValueObjects.Scenes;

// Class for scene where the biome changes in the biome atribute
public class ChangeBiomeScene : Scene
{

    // Constructor
    public ChangeBiomeScene(SceneName name, SceneDescription description, Biome biome)
        : base(name, description, biome)
    {
    }

    // To string
    public override string ToString() =>
        $"ChangeBiomeScene: {GetName()} - Description={GetDescription()}, BiomeToChange={GetBiome()}";
}

File: frontend/gameapp-frontend/src/domain/entities/scenes/change-biome-scene.js
import { Scene } from "./scene.js";

// ChangeBiomeScene
export class ChangeBiomeScene extends Scene {
    constructor(name, description, biome, id = null) {
        super(name, description, biome, id);
    }

    // setter
    setSceneName(newName) { return new ChangeBiomeScene(newName, this._description, this._biome, this._id); }
    setSceneDescription(newDescription) { return new ChangeBiomeScene(this._name, newDescription, this._biome, this._id); }
    setBiome(newBiome) { return new ChangeBiomeScene(this._name, this._description, newBiome, this._id); }



    toString() {
        return `ChangeBiomeScene(${this._id}): ${this._name} - Description=${this._description}, BiomeToChange=${this._biome}`;
    }
}